spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSqlExtensionsAstBuilder.scala

```diff
@@ -226,11 +226,13 @@ class IcebergSqlExtensionsAstBuilder(delegate: ParserInterface) extends IcebergS
     }
 
     val distributionMode = if (distributionSpec != null) {
-      DistributionMode.HASH
-    } else if (orderingSpec.UNORDERED != null || orderingSpec.LOCALLY != null) {
-      DistributionMode.NONE
+      Some(DistributionMode.HASH)
+    } else if (orderingSpec.UNORDERED != null) {
```

Comment:
Why not just return None when its not set here?  I thought that's more inline with the goal of the pr (to not override default unless explicitly set).

Reply:
I realize it would break a few tests and is a big behavior change.  So @RussellSpitzer @aokolnychyi let me know if you disagree?

Reply:
Which do you mean will break a few tests and is a big change?

Reply:
```    val distributionMode = if (distributionSpec != null) {
      DistributionMode.HASH
    } else if (orderingSpec.UNORDERED != null || orderingSpec.LOCALLY != null) {
      DistributionMode.NONE
    } else {
      DistributionMode.RANGE
    }
```

makes more sense to me as: 
```
    val distributionMode = if (distributionSpec != null) {
       Some(DistributionMode.HASH)
    } else {
      None
    }
```

Reply:
@szehon-ho I'm not sure I follow

Reply:
I thought the goal is to not set distributionMode , if its not in the ALTER statement.  Hence why I thought the above code, which returns None in those cases unless its explicitly present in ALTER statement.

The current code works for the exact statement ALTER TABLE LOCALLY UNORDERED, but not for other ALTER's that dont have distributionMode?

Reply:
I believe this function is used by other pieces of code. Specifically the Create statements where RANGE is set if there is an ordering. Maybe it makes sense to separate it out into a Distribution and ordering for ALTER and one for CREATE?

Reply:
@RussellSpitzer This method is for ALTER only.

https://github.com/apache/iceberg/blob/c16cefa5015dda417f80e0d59124cd92448787ab/spark/v3.5/spark-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4#L73

@szehon-ho The result of this method can be one of `HASH`, `NONE`, `RANGE` or unchanged depending on the combination of `distributionSpec` and `orderSpec`. Thus, we cannot just return None when `distributionSpec` is not set.